### PR TITLE
Bug/tf 208 tracking mode reverts to desktop

### PR DESCRIPTION
**Pull request by MattGrayUL**
- [x] Test plan updated (by PR creator) 
- [x] Code Review 
- [x] Developer Testing: Key item tests 
- [x] Developer Testing: Execution of the updated test plan (changed items for this work only) 
- [ ] Product Owner Review (Optional) 
- [ ] Update Release Notes on the JIRA ticket for harvest at release time

### CODE REVIEW

File: TF_Service_and_Tooling_Unity/Assets/Plugins/LeapMotion/Core/Scripts/LeapServiceProvider.cs
Comment (NatGreenwood-UL) [inline TF_Service_and_Tooling_Unity/Assets/Plugins/LeapMotion/Core/Scripts/LeapServiceProvider.cs]: Does setting this value when a coroutine is running kill the old coroutine or does it end up running headless? Concerned about having two competing with one another but not really sure how Unity Coroutines work regarding this.

Comment (MattGrayUL) [reply TF_Service_and_Tooling_Unity/Assets/Plugins/LeapMotion/Core/Scripts/LeapServiceProvider.cs]: This class is handled by XR team but to explain, the coroutine will never be running when a new one is started because it is always stopped at the beginning of this function, before this highlighted code executes

Comment (NatGreenwood-UL) [reply TF_Service_and_Tooling_Unity/Assets/Plugins/LeapMotion/Core/Scripts/LeapServiceProvider.cs]: Ah yeah I missed the Stop at the start. More for my Unity knowledge than for the sake of the code. I guess then that overwriting a coroutine _doesn't_ automatically stop the original? Good to know for future.

Comment (MattGrayUL) [reply TF_Service_and_Tooling_Unity/Assets/Plugins/LeapMotion/Core/Scripts/LeapServiceProvider.cs]: Indeed. You don't _have_ to store it as a coroutine, you can start them headlessly and they will just execute, but to handle the stopping of them you must store a reference. (or call StopAllCoroutines()).

As you guessed in your first comment, if we weren't calling StopCoroutine() above, they would all continue executing

**Review by PeterNancollis-UH — APPROVED** [review]

**Comment by NatGreenwood-UL** [comment]
I've also now tested this locally with a build from this branch and the visualiser, I think we're good to merge it into the release branch now @PeterNancollis-UH ? I'll hold off on the release testing until XR have the module update approved on their end but I can't reproduce the bug any more so I think that justifies this being in the release so we're a step closer to ready?

**Comment by PeterNancollis-UH** [comment]
@NatGreenwood-UL Agreed, I will merge it now

**Pull request merged**
